src/cli/cli.cpp

```diff
@@ -714,6 +717,28 @@ void Interpreter::ProcessDua(uint8_t aArgsLength, char *aArgs[])
 
 #endif // (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
 
+void Interpreter::ProcessAdvData(uint8_t aArgsLength, char *aArgs[])
+{
+    otError       error = OT_ERROR_NONE;
+    unsigned long oui;
+    uint8_t       advData[OT_JOINER_ADVDATA_MAX_LENGTH];
+    int           advDataLength;
+
+    VerifyOrExit(aArgsLength == 2, error = OT_ERROR_INVALID_ARGS);
+
+    SuccessOrExit(error = ParseUnsignedLong(aArgs[0], oui));
+    VerifyOrExit(oui <= std::numeric_limits<uint32_t>::max(), error = OT_ERROR_INVALID_ARGS);
+
+    advDataLength = Hex2Bin(aArgs[1], advData, sizeof(advData));
```

Comment:
How about just taking the input as text string for verification purpose? Or else we may need to take additional step to make sure the input is a legal hex string?

Reply:
> Or else we may need to take additional step to make sure the input is a legal hex string?

@lanyuwen The [Hex2Bin](https://github.com/openthread/openthread/blob/6886ff1e4c5dcf1bdc6ac7a067a81debff113938/src/cli/cli.cpp#L307) function already validated that `advData` must be a hex string (it also makes sure that the hex string does exceeds the length of the buffer). It returns `-1` in case of getting invalid arguments and we captured that error at just below.

Reply:
I see, then this looks good to me.